### PR TITLE
fix(FindableDoiRequirementsValidator); add hasCorrectPublishedYear #N…

### PR DESCRIPTION
**Pull request by torbjokv**
…P-46840

### CODE REVIEW

File: nva-datamodel-java/src/main/java/no/unit/nva/model/FindableDoiRequirementsValidator.java [hunk 1]
```diff
@@ -13,14 +16,25 @@
...
-    /* default */ static boolean meetsFindableDoiRequirements(Publication publication) {
-        return hasCorrectPublishedStatus(publication) && mandatoryFieldsAreNotNull(publication);
+    /* default */
+    static boolean meetsFindableDoiRequirements(Publication publication) {
+        return hasCorrectPublishedStatus(publication) &&
+               mandatoryFieldsAreNotNull(publication) &&
+               hasCorrectPublishedYear(publication);
+    }
+
+    private static boolean hasCorrectPublishedYear(Publication publication) {
```
Comment (brinxmat) [inline nva-datamodel-java/src/main/java/no/unit/nva/model/FindableDoiRequirementsValidator.java:34]: hasValidPublishedYear

**Review by brinxmat — CHANGES_REQUESTED** [review]
In many ways, these are not *our* requirements, but DataCite's and I'm not sure *we* care about dates in this way.

File: nva-datamodel-java/src/main/java/no/unit/nva/model/FindableDoiRequirementsValidator.java [hunk 2]
```diff
@@ -13,14 +16,25 @@
 
     private static final Set<PublicationStatus> VALID_PUBLICATION_STATUS_FOR_FINDABLE_DOI =
         Set.of(PUBLISHED, PUBLISHED_METADATA);
+    public static final int PUBLISH_YEAR_MIN = 1000;
+    public static final IntSupplier PUBLISH_YEAR_MAX = () -> Year.now().getValue() + 4;
```
[the remark this answers is not in the document]
Comment (torbjokv) [reply nva-datamodel-java/src/main/java/no/unit/nva/model/FindableDoiRequirementsValidator.java:20]: a bit of both I guess, its assumed from this documentation https://support.datacite.org/docs/field-descriptions-for-form#publication-year VALID_FUTURE_YEARS_ASSUMED?

File: nva-datamodel-java/src/main/java/no/unit/nva/model/FindableDoiRequirementsValidator.java [hunk 3]
```diff
@@ -13,14 +16,25 @@
...
+    static boolean meetsFindableDoiRequirements(Publication publication) {
+        return hasCorrectPublishedStatus(publication) &&
+               mandatoryFieldsAreNotNull(publication) &&
+               hasCorrectPublishedYear(publication);
+    }
+
+    private static boolean hasCorrectPublishedYear(Publication publication) {
+        var yearString = publication.getEntityDescription().getPublicationDate().getYear();
+        var year = attempt(() -> Integer.parseInt(yearString)).toOptional().orElse(null);
+        return nonNull(year) && year >= PUBLISH_YEAR_MIN && year <= PUBLISH_YEAR_MAX.getAsInt();
```
[the remark this answers is not in the document]
Comment (torbjokv) [reply nva-datamodel-java/src/main/java/no/unit/nva/model/FindableDoiRequirementsValidator.java:39]: yup, null here means someone passed inn "LOL" or something else that is not parsable and should validate false

**Pull request merged**